irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -37,7 +37,13 @@ namespace iroha {
 
     void SynchronizerImpl::process_commit(iroha::model::Block commit_message) {
       log_->info("processing commit");
-      auto storage = mutableFactory_->createMutableStorage();
+      auto storageResult = mutableFactory_->createMutableStorage();
+      std::unique_ptr<ametsuchi::MutableStorage> storage;
+      storageResult.match(
+          [&](expected::Value<std::unique_ptr<ametsuchi::MutableStorage>> &_storage) {
+            storage = std::move(_storage.value);
+          }, [](expected::Error<std::string> &error) {}
```

Comment:
I don't think it is good idea to leave error branch empty. At least throw something.

Reply:
error is processed later: 
 ```
storageResult.match(
          [&](expected::Value<std::unique_ptr<ametsuchi::MutableStorage>> &_storage) {
            storage = std::move(_storage.value);
          }, [](expected::Error<std::string> &error) {}
      );
if (not storage) {
        log_->error("Cannot create mutable storage");
        return;
      }
```

I am not sure, how it should be rewritten